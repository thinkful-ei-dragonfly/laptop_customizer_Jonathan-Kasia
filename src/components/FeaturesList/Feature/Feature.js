import React from 'react'

function Feature(props){
  console.log(props)
  return <li className="feature__item">
  <div className={props.featureClass}
    
    onClick={e => props.updateFeature(props.key, props.item)}>
      { props.item.name }
      ({ new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
        .format(props.item.cost) })
  </div>
</li>
}

export default Feature