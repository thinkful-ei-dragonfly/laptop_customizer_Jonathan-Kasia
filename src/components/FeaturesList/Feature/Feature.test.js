import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Feature from './Feature'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import { isMainThread } from 'worker_threads';

describe('Feature Component', () => {
  it("Renders without crashing", () =>{
    const div = document.createElement('div')
    ReactDOM.render(<Feature />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it("Renders the UI as expected", () =>{
    const tree = renderer
    .create(<Feature />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

})