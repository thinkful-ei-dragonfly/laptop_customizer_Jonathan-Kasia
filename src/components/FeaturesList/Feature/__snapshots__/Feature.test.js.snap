// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Component Renders UI as expected when feature is not selected 1`] = `
<li
  className="feature__item"
>
  <div
    className="feature__option"
    onClick={[Function]}
  >
    17th Generation Intel Core HB (7 Core with donut spare)
    (
    $700.00
    )
  </div>
</li>
`;

exports[`Feature Component Renders UI as expected when feature is selected 1`] = `
<li
  className="feature__item"
>
  <div
    className="feature__option feature__selected"
    onClick={[Function]}
  >
    17th Generation Intel Core HB (7 Core with donut spare)
    (
    $700.00
    )
  </div>
</li>
`;

exports[`Feature Component Renders the UI as expected 1`] = `
<li
  className="feature__item"
>
  <div
    className="feature__option"
    onClick={[Function]}
  >
    17th Generation Intel Core HB (7 Core with donut spare)
    (
    $700.00
    )
  </div>
</li>
`;
